src/components/OneArticle.jsx

```diff
@@ -28,6 +30,59 @@ function OneArticle() {
       });
   }, [article_id]);
 
+  function handleClick(vote, article_id) {
+    const updateVote = { ...singleArticle };
+
+    if (!downvoteClick && !upvoteClick) {
+      updateVote.votes += vote;
+      setSingleArticle(updateVote);
+
+      axios
+        .patch(
+          `https://backend-service-project-2.onrender.com/api/articles/${article_id}`,
+          { inc_votes: vote }
+        )
+        .catch((err) => {
+          updateVote.votes -= vote;
+          setSingleArticle(updateVote);
+        });
+    } else if (!downvoteClick && upvoteClick) {
+      updateVote.votes -= 2;
+      setSingleArticle(updateVote);
+
+      axios
+        .patch(
+          `https://backend-service-project-2.onrender.com/api/articles/${article_id}`,
+          { inc_votes: -2 }
+        )
+        .catch((err) => {
+          updateVote.votes += 2;
+          setSingleArticle(updateVote);
+        });
+    } else if (downvoteClick && !upvoteClick) {
+      updateVote.votes += 2;
+      setSingleArticle(updateVote);
+
+      axios
+        .patch(
+          `https://backend-service-project-2.onrender.com/api/articles/${article_id}`,
+          { inc_votes: 2 }
+        )
+        .catch((err) => {
+          updateVote.votes -= 2;
+          setSingleArticle(updateVote);
+        });
+    }
+
+    if (vote === 1) {
+      setUpvoteClick(true);
+      setDownvoteClick(false);
+    } else if (vote === -1) {
+      setUpvoteClick(false);
+      setDownvoteClick(true);
+    }
+  }
+

```

Comment:
This vote button works well using optimistic rendering when the internet is reliable 🙌 

If you open this page, then lose the internet connection (you can do this in the developer tools by going to the Network tag and changing 'No throttling' to 'Offline') the votes change their behaviour (upvoting sometimes gives you more votes than it should) and the state doesn't correct itself to reflect the patch request has failed.